docs/testplan-style.md

```diff
@@ -0,0 +1,243 @@
+# Feature Profiles Test Plan Style Guide
+
+A test plan is a Markdown file that describes the step-by-step procedure about a
+test. It is typically found as the `README.md` in the test directory.
+
+For example:
+
+*   `feature/interface/singleton/ate_tests/singleton_test/README.md` documents
+    the test plan for the issue
+    [RT-5.1 Singleton Interface](https://github.com/openconfig/featureprofiles/issues/111).
+*   `feature/interface/singleton/ate_tests/singleton_test/singleton_test.go`
+    implements the issue.
+*   `feature/interface/aggregate/ate_tests/aggregate_test/README.md` documents
+    the test plan for the issue
+    [RT-5.2 Aggregate Interface](https://github.com/openconfig/featureprofiles/issues/112).
+*   `feature/interface/aggregate/ate_tests/aggregate_test/aggregate_test.go`
+    implements the issue.
+
+## Test Plan Template
+
+The test plan in `README.md` is generally structured like this:
+
+```
+# RT-5.1: Singleton Interface
+
+## Summary
+
+[Insert a 1-3 sentence description of the test.]
+
+## Procedure
+
+[Write an ordered list of steps.  Each step may be given a
+descriptive name.]
+
+1. Step 1
+2. Step 2
+3. ...
+
+## Config Parameter Coverage
+
+[Write a list of OpenConfig paths which are expected to be included.]
+
+*   /interfaces/interface/config/name
+*   /interfaces/interface/config/description
+*   ...
+
+## Telemetry Parameter Coverage
+
+[Write a list of OpenConfig paths which are expected to be included.]
+
+*   /interfaces/interface/state/oper-status
+*   /interfaces/interface/state/admin-status
+*   ...
+```
+
+## Development Approach for Tests
+
+### Vendor Neutrality
+
+Test plans should use a vendor-neutral mechanism to configure the DUT, e.g. in
+the OpenConfig public model or other OpenConfig protocols such as gNMI, gRIBI,
+and gNOI. It should depend only on the behavior of the open specification.
+
+### Device/Topology Neutrality
+
+Tests are to be built against a minimal topology that consists of a single
+device (referred to as DUT) against a reference implementation (referred to as
+an ATE) of routing protocols and data plane functionality.
+
+Each test plan must choose from one of the following testbeds and adhere to it
+for the entirety of the test plan:
+
+*   atedut\_2: two interconnected ports between ATE and DUT.
+*   atedut\_4: four interconnected ports between ATE and DUT.
+*   atedut\_12: 4 singleton ports and 8 breakout ports interconnected between
+    ATE and DUT. Ondatra testbed models the logical ports, but the test is
+    responsible for activating the breakout.
+*   dutdut: four interconnected ports between two DUTs.
```

Comment:
Having `dutdut` here is perhaps not consistent with the need to have single DUT, single ATE tests as defined above.

I wonder also whether being prescriptive about these topologies is assuming that one knows what the test cases that are being written are. Is there any reason that someone couldn't write a test that has 32, 64, 96, ... ports available to it - for e.g., scaling testing.